pkg/distributors/contracts/MerkleOrchard.sol

```diff
@@ -63,16 +63,57 @@ contract MerkleOrchard {
     ) internal {
         uint256[] memory amounts = new uint256[](tokens.length);
 
+        // To save gas when setting claimed statuses in storange we group updates
+        // into currentBits for a particular channel, only setting them when a claim
+        // on a new channel is seen, or on the final iteration
+
+        // for aggregating claims
+        uint256 currentBits;
+        bytes32 currentChannelId;
+        uint256 currentWordIndex;
+        uint256 currentClaimAmount;
+
         Claim memory claim;
-        IERC20 token;
         for (uint256 i = 0; i < claims.length; i++) {
             claim = claims[i];
-            token = tokens[claim.tokenIndex];
 
-            require(!isClaimed(token, claim.distributor, claim.distribution, liquidityProvider), "cannot claim twice");
+            // When we process a new claim we either
+            // a) aggregate the new claim bit with previous claims of the same channel/claim bitmap
+            // b) set claim status and start aggregating a new set of currentBits for a new channel/word
+            if (currentChannelId == bytes32(0)) {
+                currentChannelId = keccak256(abi.encodePacked(address(tokens[claim.tokenIndex]), claim.distributor));
+                currentWordIndex = claim.distribution / 256;
+                currentBits = 1 << claim.distribution % 256;
+                currentClaimAmount = claim.balance;
+            } else if (
+                currentChannelId == keccak256(abi.encodePacked(address(tokens[claim.tokenIndex]), claim.distributor))
+            ) {
+                if (currentWordIndex == claim.distribution / 256) {
+                    currentBits |= 1 << claim.distribution % 256;
+                    currentClaimAmount += claim.balance;
+                } else {
+                    _setClaimedBits(liquidityProvider, currentChannelId, currentWordIndex, currentBits);
+
+                    currentWordIndex = claim.distribution / 256;
+                    currentBits = 1 << claim.distribution % 256;
+                }
```

Comment:
```suggestion
                currentClaimAmount += claim.balance;
                if (currentWordIndex == claim.distribution / 256) {
                    currentBits |= 1 << claim.distribution % 256;
                } else {
                    _setClaimedBits(liquidityProvider, currentChannelId, currentWordIndex, currentBits);

                    currentWordIndex = claim.distribution / 256;
                    currentBits = 1 << claim.distribution % 256;
                }
```

Currently if you go into the next word of the channel then it looks like you won't get any of the tokens.